napari/layers/shapes/_accelerated_triangulate.py

```diff
@@ -0,0 +1,510 @@
+from __future__ import annotations
+
+import numpy as np
+from numba import njit
+
+
+@njit(cache=True, inline='always')
+def _calc_output_size(
+    normals: np.ndarray, closed: bool, cos_limit: float, bevel: bool
+) -> int:
+    """Calculate the size of the output arrays for the triangulation.
+
+    Parameters
+    ----------
+    normals : np.ndarray
+        Nx2 array of normal vectors of the path
+    closed : bool
+        Bool if shape edge is a closed path or not.
+    cos_limit : float
+        Miter limit which determines when to switch from a miter join to a
+        bevel join
+    bevel : bool
+        Bool which if True causes a bevel join to always be used.
+        If False a bevel join will only be used when the miter limit is exceeded
+
+    Returns
+    -------
+    int
+        number of points in the output arrays
+
+    Notes
+    -----
+    we use cos_limit instead of maximum miter length
+    for performance reasons.
+    This is an equivalent check, see note in generate_2D_edge_meshes
+    """
+    point_count = len(normals) * 2
+    if closed:
+        point_count += 2
+
+    bevel_count = 0
+
+    if bevel:
+        bevel_count = len(normals)
+        if not closed:
+            bevel_count -= 2
+    else:
+        for i in range(1, len(normals) - 1):
+            cos_angle = (
+                normals[i - 1, 0] * normals[i, 0]
+                + normals[i - 1, 1] * normals[i, 1]
+            )
+            if cos_limit > cos_angle:
```

Comment:
```suggestion
            if cos_angle < cos_limit:
```
Subtle change but flipping the check to logically read: If calculated cos_angle (which changes in the else block) is less than cos_limit (which doesn't change in this function).